packages/bvaughn-architecture-demo/src/suspense/FrameStepsCache.ts

```diff
@@ -0,0 +1,33 @@
+import { FrameId, PauseId, PointDescription } from "@replayio/protocol";
+import cloneDeep from "lodash/cloneDeep";
+
+import { ReplayClientInterface } from "shared/client/types";
+
+import { createGenericCache2 } from "./createGenericCache";
+import { updateMappedLocation } from "./PauseCache";
+
+export const {
+  getValueSuspense: getFrameStepsSuspense,
+  getValueAsync: getFrameStepsAsync,
+  getValueIfCached: getFrameStepsIfCached,
+} = createGenericCache2<
+  ReplayClientInterface,
+  [pauseId: PauseId, frameId: FrameId],
+  PointDescription[] | undefined
+>(
+  async (client, pauseId, frameId) => {
+    try {
+      const frameSteps = await client.getFrameSteps(pauseId, frameId);
+      const updatedFrameSteps = frameSteps.map(frameStep => cloneDeep(frameStep));
```

Comment:
Just kind of curious if this would work?
```suggestion
const updatedFrameSteps = cloneDeep(frameSteps);
```
No need to change it regardless.